samples/httpd/main.c

```diff
@@ -85,20 +85,17 @@ int main(void)
 	packet_timer(NULL);
 
 #if USE_DHCP
-	debugPrint("Waiting for DHCP...\n");
-	while (dhcp_supplied_address(g_pnetif) == 0)
-		NtYieldExecution();
-	debugPrint("DHCP bound!\n");
+	nxLogPrint("Waiting for DHCP...\n");
+	while (dhcp_supplied_address(g_pnetif) == 0) SwitchToThread();
+
+	nxLogPrint("DHCP bound!\n");
 #endif
 
-	debugPrint("\n");
-	debugPrint("IP address.. %s\n", ip4addr_ntoa(netif_ip4_addr(g_pnetif)));
-	debugPrint("Mask........ %s\n", ip4addr_ntoa(netif_ip4_netmask(g_pnetif)));
-	debugPrint("Gateway..... %s\n", ip4addr_ntoa(netif_ip4_gw(g_pnetif)));
-	debugPrint("\n");
+	nxLogPrintf("\nIP address.. %s\nMask........ %s\nGateway..... %s\n\n", ip4addr_ntoa(netif_ip4_addr(g_pnetif)), ip4addr_ntoa(netif_ip4_netmask(g_pnetif)), ip4addr_ntoa(netif_ip4_gw(g_pnetif)));
 
 	http_server_netconn_init();
-	while (1) NtYieldExecution();
+	while (1) SwitchToThread();
```

Comment:
I'm fine with this change, but it should not be part of this PR as it is unrelated to the functionality it adds.